SingleCellPatch/instance_clustering.py

```diff
@@ -85,7 +85,7 @@ def instance_clustering(cell_segmentation,
 
     """
     cell_segmentation = check_segmentation_dim(cell_segmentation)
-    all_cells = cell_segmentation[0, 0] < fg_thr
+    all_cells = np.mean(cell_segmentation[0], axis=0) < fg_thr
```

Comment:
Sorry.  Just to elaborate on what this does.  `cell_segmentation` is of shape `(n_classes(3), z(1), x, y)` .  So your adjustment says to take `class=0`, average across z, then threshold this averaged image based on `fg_thr` ?

Reply:
@bryantChhun That's correct. The segmentation mask stack can have single or multiple z slices depending on the input. Averaging along z will make sure both cases are handled